lib/components/Forms/Fields/NumberInput/index.tsx

```diff
@@ -0,0 +1,118 @@
+import { Icon } from "@/components/Utilities/Icon"
+import { ChevronDown, ChevronUp } from "@/icons"
+import { Input } from "@/ui/input"
+import { forwardRef, useEffect, useState } from "react"
+import { InputProps } from "../Input"
+
+const FORMAT = /^([0-9]+)$/
+// eslint-disable-next-line no-useless-escape
+const FORMAT_WITH_DECIMALS = /^([0-9]+)(?:[\.,]([0-9]+)?)?$/
```

Comment:
This numberinput doesn't support negative numbers. I have a [PR ](https://github.com/factorialco/factorial/pull/49953)to introduce negative support to NumberField in gamma, if it's accepted perhaps you could port that logic to this component.

Reply:
Merged the PR mentioned, that should help simplify the useEffect need as well.

Reply:
Thanks! I ported that logic to here 🙂